libs/platforms/rosa/terraform/terraform.py

```diff
@@ -0,0 +1,262 @@
+            )
+            check_code, check_out, check_err = self.utils.subprocess_exec(
+                "rosa describe cluster -c " + cluster_name, log_output=False
+            )
+            if check_code != 0:
+                cluster_info["status"] = "deleted"
+            else:
+                cluster_info["status"] = "not deleted"
+        else:
+            cluster_info["status"] = "not deleted"
+        cluster_end_time = int(datetime.datetime.utcnow().timestamp())
+        cluster_info["destroy_duration"] = cluster_delete_end_time - cluster_start_time
+        cluster_info["destroy_all_duration"] = cluster_end_time - cluster_start_time
+        try:
+            with open(cluster_info['path'] + "/metadata_destroy.json", "w") as metadata_file:
+                json.dump(cluster_info, metadata_file)
+        except Exception as err:
+            self.logging.error(err)
+            self.logging.error(f"Failed to write metadata_install.json file located at {cluster_info['path']}")
+        if self.es is not None:
+            cluster_info["timestamp"] = datetime.datetime.utcnow().isoformat()
+            self.es.index_metadata(cluster_info)
+
+    def get_workers_ready(self, kubeconfig, cluster_name):
+        super().get_workers_ready(kubeconfig, cluster_name)
+        myenv = os.environ.copy()
+        myenv["KUBECONFIG"] = kubeconfig
+        self.logging.info(f"Getting node information for Terraform installed cluster {cluster_name}")
+        nodes_code, nodes_out, nodes_err = self.utils.subprocess_exec("oc get nodes -o json", extra_params={"env": myenv, "universal_newlines": True}, log_output=False)
+        try:
+            nodes_json = json.loads(nodes_out)
+        except Exception as err:
+            self.logging.debug(f"Cannot load command result for cluster {cluster_name}")
+            self.logging.debug(err)
+            return 0
+        nodes = nodes_json["items"] if "items" in nodes_json else []
+        status = []
+        for node in nodes:
+            labels = node.get("metadata", {}).get("labels", {})
+            if "node-role.kubernetes.io/worker" in labels and "node-role.kubernetes.io/control-plane" not in labels and "node-role.kubernetes.io/infra" not in labels:
+                conditions = node.get("status", {}).get("conditions", [])
+                for condition in conditions:
+                    if "type" in condition and condition["type"] == "Ready":
+                        status.append(condition["status"])
+        status_list = {i: status.count(i) for i in status}
+        ready_nodes = status_list["True"] if "True" in status_list else 0
+        return ready_nodes
+
+    def create_cluster(self, platform, cluster_name):
+        super().create_cluster(platform, cluster_name)
+        cluster_info = platform.environment["clusters"][cluster_name]
+        cluster_info["uuid"] = self.environment["uuid"]
+        cluster_info["hostedclusters"] = self.environment["cluster_count"]
```

Comment:
TF does not support hypershift yet

Reply:
@morenod I guess it's just a placeholder var

Reply:
yes, I copied from the hypershift installer, that var makes no sense on terraform installations

Reply:
deleted